src/helpers/DataRequirementHelpers.ts

```diff
@@ -35,22 +37,82 @@ export function flattenFilters(filter: AnyFilter): AnyFilter[] {
  * @param filter the filter to translate
  * @returns codeFilter to be put on the DataRequirement
  */
-export function generateDetailedCodeFilter(filter: EqualsFilter | InFilter): R4.IDataRequirement_CodeFilter | null {
+export function generateDetailedCodeFilter(filter: EqualsFilter | InFilter, dataType: string): R4.IDataRequirement_CodeFilter | null {
+
+
+  const codeLookup = (dataType: string, attribute: string): (string | undefined) => {
+
+    const validDataTypes : string[] = ["Observation", "Procedure", "Encounter", "MedicationRequest"];
+    
+    if (!(validDataTypes.includes(dataType))) {
+      return undefined;
+    }
+
+    else if (dataType === "Observation" && attribute === "status") {
+      return "http://hl7.org/fhir/observation-status";
+    }
+    
+    else if (dataType === "Procedure" && attribute === "status") {
+      return "http://hl7.org/fhir/event-status";
+    }
+
+    else if (dataType === "Encounter" && attribute === "status") {
+      return 'http://hl7.org/fhir/encounter-status';
+    }
+
+    else if (dataType === 'MedicationRequest'){
+      
+      switch(attribute){
+        case 'status':
+          return "http://hl7.org/fhir/CodeSystem/medicationrequest-status";
+          
+        case 'intent':
+          return "http://hl7.org/fhir/CodeSystem/medicationrequest-intent";
+            
+        case 'priority':
+          return "http://hl7.org/fhir/request-priority";
+          
+        default:
+          return undefined;
+      }
+    }
+    return undefined;
+    
+  }
+  let system : string | undefined = codeLookup(dataType, filter.attribute);
```

Comment:
This thing is a small nitpick, so I might defer to whatever you prefer. BUT

In my opinion, it's better to use `null` over `undefined` to indicate when something has "No value" versus the concept of it "not existing." Obviously these two things are very similar, but I recommend using `null` instead of `undefined` here. This would require switching up the return values and types, but it shouldn't affect much else